admin-scripts/extract_mini_repository.sh

```diff
@@ -0,0 +1,118 @@
+#! /bin/sh
+
+set -e
+
+if [ $# = 0 ]; then
+   cat <<EOF
+
+This script is able to generate a minimalistic and self-contained 
+opam-repository for a given set of packages and compiler versions.
+The resulting repository will be in './opam-mini-repository.tar.gz'
+
+Example:
+
+ ## Including only the latest compiler version:
+ $0 ocp-indent lwt
+
+ ## Including a specific verion of OCaml or multiple version.
+ $0 ocaml.4.02.1 ocaml.4.01.0 ocp-indent lwt
+
+EOF
+   exit 1
+fi
+
+DEFAULT_COMPILER=4.02.1
+OPAM_REPO=https://github.com/ocaml/opam-repository/archive/master.tar.gz
+
+TARGET=$(pwd)
+DIR=$(mktemp -d)
+REPO_NAME=opam-repository-master
+REPO=$DIR/${REPO_NAME}
+REQUEST=$@
+COMPILERS=
+
+for package in $REQUEST; do
+    case $package in
+        ocaml\.*)
+            COMPILERS="$COMPILERS ${package#ocaml.}"
+            ;;
+        *)
+            PACKAGES="$PACKAGES $package"
+            ;;
+    esac
+done
+
+if [ -z "$COMPILERS" ]; then
+    COMPILERS="$DEFAULT_COMPILER"
+fi
+
+cleanup() {
+    rm -r $DIR
+}
+
+trap cleanup 0
+
+## Fetch the opam-repository
+
+cd $DIR
+wget $OPAM_REPO
+tar xf master.tar.gz
+
+cd $REPO
+rm -rf $REPO/.git
+
+## Remove the unrequired compilers description
+
+unrequired_compiler() {
+  for version in $COMPILERS; do
+      if [ ${version} = $1 ]; then return 1; fi
+  done
+  return 0
+}
+
+for dir in $(ls -d compilers/*/*); do
+    if unrequired_compiler ${dir##compilers/*/}; then
+        rm -r $dir
+    fi
+done
+
+rmdir --ignore-fail-on-non-empty compilers/*
+
+## Convert the required compilers as packages
+
+$TARGET/compilers-to-packages.ml
+
+## Fetch the packages and compilers archives
+
+for version in ${COMPILERS}; do
+    opam admin make --resolve --compiler $version ocaml.$version $PACKAGES
+done
+
+## Remove the unrequired packages description
+
+ARCHIVES=$(ls $REPO/archives)
+
+unrequired_version() {
+    case $1 in
+        base-*)
+            return 1;;
+        *)
+            for version in $ARCHIVES; do
+                if [ ${version%+opam.tar.gz} = $1 ]; then return 1; fi
+            done
+    esac
+    return 0
+}
+
+for dir in $(ls -d packages/*/*); do
+    if unrequired_version ${dir##packages/*/}; then
```

Comment:
is this substitution a bash-ism? You've used `/bin/sh` at the top of the script.